homeassistant/components/homewizard/sensor.py

```diff
@@ -414,16 +411,80 @@ class HomeWizardSensorEntityDescription(SensorEntityDescription):
 )
 
 
+EXTERNAL_SENSORS = {
+    ExternalDevice.DeviceType.GAS_METER: HomeWizardExternalSensorEntityDescription(
+        key="gas_meter",
+        translation_key="total_gas_m3",
+        suggested_device_class=SensorDeviceClass.GAS,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_name="Gas meter",
+    ),
+    ExternalDevice.DeviceType.HEAT_METER: HomeWizardExternalSensorEntityDescription(
+        key="heat_meter",
+        translation_key="total_energy_gj",
+        suggested_device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_name="Heat meter",
+    ),
+    ExternalDevice.DeviceType.WARM_WATER_METER: HomeWizardExternalSensorEntityDescription(
+        key="warm_water_meter",
+        translation_key="total_liter_m3",
+        suggested_device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_name="Warm water meter",
+    ),
+    ExternalDevice.DeviceType.WATER_METER: HomeWizardExternalSensorEntityDescription(
+        key="water_meter",
+        translation_key="total_liter_m3",
+        suggested_device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_name="Water meter",
+    ),
+    ExternalDevice.DeviceType.INLET_HEAT_METER: HomeWizardExternalSensorEntityDescription(
+        key="inlet_heat_meter",
+        translation_key="total_energy_gj",
+        suggested_device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_name="Inlet heat meter",
+    ),
+}
+
+
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Initialize sensors."""
     coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities(
+
+    # Migrate original gas meter sensor to ExternalDevice
```

Comment:
Please add code which removes the now defunct `gas_unique_id` sensor, otherwise users will still see it, but unavailable

Reply:
Good point! Can you hint me what function to use for that or an integration that does this?

Reply:
Sure, here's an example https://github.com/home-assistant/core/blob/95fdae5e9d050e1a71781ad00a7a8b3046b1c0fd/homeassistant/components/accuweather/__init__.py#L52-L58

Reply:
Cool, added removal and test.